packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -395,10 +395,10 @@ class BorderRadius extends BorderRadiusGeometry {
   RRect toRRect(Rect rect) {
     return RRect.fromRectAndCorners(
       rect,
-      topLeft: topLeft,
-      topRight: topRight,
-      bottomLeft: bottomLeft,
-      bottomRight: bottomRight,
+      topLeft: topLeft.clamp(minimum: Radius.zero), // ignore_clamp_double_lint
```

Comment:
Is this missing the same TODO present in the other files? Or is this one ... different?

Reply:
Actually, yes, these are different. Even though I told Hans above that they'll be removed, that's wrong. These will stay in place, since the new code will assert if negative radii are passed in, and we want to clamp here.

Reply:
Let's leave a comment then explaining why the ignores here are permanent and legit.

Reply:
Actually, does it ever make sense to have a BorderRadius with negative radii that would require clamping here? Or should we already assert non-negativity when a BorderRadius is created? If negativity is legal, should this method just assert if you are trying to create a rect (illegal with negative radii) from it? Feels strange that this method just clamps the negativity away?

Reply:
I thought about that: `BorderRadius` is a collection of radii, and it is useful for radii to be negative, since you could be composing them with other radii to end up with a positive result.  `RRect`s, on the other hand, aren't meaningful with negative radii.

We could assert if any are negative, but I thought that clamping would be less breaking.

Reply:
Makes sense. Maybe add this clamping behavior to the doc comment so its less of a surprise?

Reply:
Done.